Sources/ManageSmileys.php

```diff
@@ -1011,6 +1013,58 @@ function EditSmileys()
 				if (empty($_POST['smiley_code']))
 					fatal_lang_error('smiley_has_no_code');
 
+				// If upload a new smiley image, check that smiley set folders are writable for the sets with new images.
+				$writeErrors = array();
+				foreach ($_FILES['smiley_upload']['name'] as $set => $name)
+				{
+					if (!empty($name))
+						if (!is_writable($context['smileys_dir'] . '/' . $set))
+							$writeErrors[] = $set['path'];
+				}
+
+				if (!empty($writeErrors))
+					fatal_lang_error('smileys_upload_error_notwritable', true, array(implode(', ', $writeErrors)));
```

Comment:
General question to whomever: should these errors be logged? (imo they shouldn't, but there may be a reason here that I missed)

Reply:
I did also wonder about that. This seems to be used as error handling in many places in the admin section. Is there another way that will just show the error to the user?
The code is mostly copied from AddSmiley()